plugins/org.yakindu.sct.model.sexec/src/org/yakindu/sct/model/sexec/concepts/EventProcessing.xtend

```diff
@@ -0,0 +1,382 @@
+		else _empty	
+	}
+
+
+	def Step _clearInEvents(ExecutionFlow it) {
+		if ( clearInEvents !== null) clearInEvents._call._statement 
+		else _empty	
+	}
+
+	def Step _clearInternalEvents(ExecutionFlow it) {
+		if ( clearInternalEvents !== null) clearInternalEvents._call._statement 
+		else _empty	
+	}
+
+	def Step _swapIncomingEvents(ExecutionFlow it) {
+		if (swapIncomingEvents !== null)
+			swapIncomingEvents._call._statement
+		else _empty
+	}
+
+	def Step _swapInternalEvents(ExecutionFlow it) {
+		if (swapInternalEvents !== null)
+			swapInternalEvents._call._statement
+		else _empty
+	}
+
+
+	def Step _clear(EObject it){
+		_conceptSequence(CLEAR_EVENT, it)	
+	}
+	
+	def Step _move(EObject source, EObject target){
+		_conceptSequence(MOVE_EVENT, source, target)	
+	}
+	
+	
+	def transformEventAccess(ExecutionFlow flow) {
+	
+		if ( ! flow.hasEventBuffer ) return 
+		
+		val bufferedEvents = #[flow.incomingEvents, flow.timeEvents, flow.localEvents].flatten.toSet
+
+		val allEventReferences = flow.eAllContents
+										.filter(Expression)
+										.filter[ e | 
+											bufferedEvents.contains(e.referenceOrFeature)
+											&& !(e.eContainer instanceof FeatureCall)
+											&& !(e.eContainer instanceof EventRaisingExpression)
+										]
+										
+		val HashMap<EObject, Expression> allEventAccessExpression = newHashMap
+		flow.bufferEventExpressions.forEach[ e | if (e.referenceOrFeature !== null) allEventAccessExpression.put(e.referenceOrFeature, e)]
+													
+		allEventReferences.forEach[ expression | 
+			val event = expression.referenceOrFeature as Event
+			val bufferEvent = event.createBufferEvent
+			val bufferEventExpression = EcoreUtil.copy(allEventAccessExpression.get(bufferEvent))
+			
+			if ( expression.eContainer instanceof EventValueReferenceExpression ) {
+				expression.eContainer.eContainer.substitute(expression.eContainer, bufferEventExpression._meta(bufferEvent.valueFeature))
+			} else {
+				expression.eContainer.substitute(expression, bufferEventExpression)			
+			}
+		]
+				
+	}
+	
+	/* Substitutes the EObject oldValue which is contained in EObject parent by EObject newValue.
+	 * If oldValue is not contained in parent then nothing happens. 
+	 */
+	protected def substitute(EObject parent, EObject oldValue, EObject newValue) {
+		if (parent !== null ) {
+			val ref = parent.eClass.getEAllStructuralFeatures
+				.filter(EReference)
+				.filter[ f | 
+					f.isChangeable 
+					&& !f.isDerived 
+					&& f.isContainment
+					&& parent.eIsSet(f)
+				]
+				.findFirst[ f | 
+					if (f.isMany) {
+						(parent.eGet(f) as List<EObject>).contains(oldValue)
+					} else {
+						parent.eGet(f) === oldValue
+					}
+				]
+			
+			if ( ref !== null ) {
+				if (ref.isMany) {
+					val list = parent.eGet(ref) as List<EObject>
+					list.set(list.indexOf(oldValue), newValue)
+				}
+				else parent.eSet(ref, newValue)				
+			}			
+		}	
+	}
```

Comment:
Why not using `EcoreUtil.replace(o1,o2)` ?

Reply:
It is so nice to do it on my own ;-) 
.... changed